services/hooks/src/listeners.js

```diff
@@ -76,31 +80,29 @@ class HookListeners {
     }, async ({payload}) => {
       // Get a fresh copy of the hook and fire it, if it still exists
       let latestHook = await this.Hook.load({
-        hookGroupId: hook.hookGroupId,
-        hookId: hook.hookId,
+        hookGroupId: hookGroupId,
+        hookId: hookId,
       }, true);
       if (latestHook) {
         await this.taskcreator.fire(latestHook, {firedBy: 'pulseMessage', payload});
       }
     });
 
-    this.listeners.push(listener);
+    this.listeners[queueName] = listener;
   }
 
   /** Delete a listener for the given queueName  */
   async removeListener(queueName) {
     debug(`${queueName}: stop listening`);
-    let removeIndex = this.listeners.findIndex(({_queueName}) => _queueName === queueName);
-    if (removeIndex > -1) {
-      const listener = this.listeners[removeIndex];
+    const listener = this.listeners[queueName];
+    delete this.listeners[queueName];
+    if (listener) {
       await listener.stop();
-      this.listeners.splice(removeIndex, 1);
     }
   }
 
   haveListener(queueName) {
-    const index = this.listeners.findIndex(({_queueName}) => _queueName === queueName);
-    return index > -1;
+    return !!this.listeners[queueName];
```

Comment:
Isn't `if (!this.haveListener(queue.queueName))` the same as `if (!this.listeners[queueName])`?

Reply:
Yeah, that might be simpler.